File: tests/class/fail-init.swift
public class Possible {
	let foo: Int = 0
	init?(successfully: Bool) {
		if (!successfully) {
			return nil
		}
	}
	init() {
	}
}

public func allocate(successfully: Bool) -> Possible? {
	return Possible(successfully: successfully)
}

public func allocateAlways() -> Possible? {
	return Possible()
}

File: tests/class/fail-init.js
export function allocate$successfully$(successfully) {
  const possible = new Possible();
  possible.foo = 0;

  if (!successfully) {
    return null;
  }

  return possible;
}
export function allocateAlways() {
  const possible0 = new Possible();
  possible0.foo = 0;
  return possible0;
}
export class Possible {}